test/e2e/basic_test.go

```diff
@@ -130,6 +148,39 @@ func TestPodLifecycle(t *testing.T) {
 	if _, err := findPodInPodStats(stats, pod1); err == nil {
 		t.Fatalf("expected to NOT find pod \"%s/%s\" in the slice of pod stats", pod1.Namespace, pod1.Name)
 	}
+
+	// Wait for the "nginx-0-X" pod to be deleted in a separate goroutine.
+	// This ensures that we don't possibly miss the MODIFIED/DELETED events due to establishing the watch too late in the process.
+	pod0Ch := make(chan struct{})
+	go func() {
+		// Wait for the "nginx-0-X" pod to be reported as having been deleted.
+		if err := f.WaitUntilPodDeleted(pod0.Namespace, pod0.Name); err != nil {
+			t.Fatal(err)
+		} else {
+			// Close the pod0Ch channel, signaling we've observed deletion of the pod.
+			close(pod0Ch)
+		}
+	}()
+
+	// Forcibly delete the "nginx-0" pod.
+	if err := f.DeletePodImmediately(pod0.Namespace, pod0.Name); err != nil {
+		t.Fatal(err)
+	}
+	// Wait for the delete event to be ACKed.
+	<-pod0Ch
+	// Give the provider some time to react to the MODIFIED/DELETED events before proceeding.
+	time.Sleep(deleteGracePeriodForProvider)
```

Comment:
The mock provider will react immediately. We just need a small delay (and it could probably be smaller than this) so that the event is processed. Whatever interval we'd use in a loop, we'd most probably succeed after the very first attempt.

Reply:
OK, I'll take note and think about this in my (wip) plan for #368.